core/src/test/java/org/apache/brooklyn/entity/group/SequenceGroupTest.java

```diff
@@ -0,0 +1,165 @@
+import com.google.common.collect.ImmutableSet;
+
+public class SequenceGroupTest extends BrooklynAppUnitTestSupport {
+
+    private TestApplication app;
+    private SequenceGroup group;
+    private TestEntity e1, e2, e3;
+
+    @BeforeMethod(alwaysRun=true)
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+
+        app = TestApplication.Factory.newManagedInstanceForTests();
+        group = app.createAndManageChild(EntitySpec.create(SequenceGroup.class)
+                .configure(SequenceGroup.SEQUENCE_STRING_SENSOR, Sensors.newStringSensor("test.sequence"))
+                .configure(SequenceGroup.SEQUENCE_FORMAT, "test-%02d"));
+        e1 = app.createAndManageChild(EntitySpec.create(TestEntity.class));
+        e2 = app.createAndManageChild(EntitySpec.create(TestEntity.class));
+        e3 = app.createAndManageChild(EntitySpec.create(TestEntity.class));
+    }
+
+    @AfterMethod(alwaysRun=true)
+    @Override
+    public void tearDown() throws Exception {
+        if (app != null) Entities.destroyAll(app.getManagementContext());
+
+        super.tearDown();
+    }
+
+
+    @Test
+    public void testGroupDefaults() throws Exception {
+        assertTrue(group.getMembers().isEmpty());
+    }
+
+    @Test
+    public void testGroupWithMatchingFilterReturnsOnlyMatchingMembers() throws Exception {
+        group.setEntityFilter(EntityPredicates.idEqualTo(e1.getId()));
+
+        assertEqualsIgnoringOrder(group.getMembers(), ImmutableList.of(e1));
+        assertAttributeEquals(e1, SequenceGroup.SEQUENCE_VALUE, 1);
+        assertAttributeEquals(group, SequenceGroup.SEQUENCE_NEXT, 2);
+    }
+
+    @Test
+    public void testGroupConfiguration() throws Exception {
+        group.setEntityFilter(EntityPredicates.idEqualTo(e1.getId()));
+
+        assertEqualsIgnoringOrder(group.getMembers(), ImmutableList.of(e1));
+        assertAttributeEquals(e1, SequenceGroup.SEQUENCE_STRING, null);
+        assertAttributeEquals(e1, Sensors.newStringSensor("test.sequence"), "test-01");
+    }
+
+    @Test
+    public void testAlternateGroupConfiguration() throws Exception {
+        AttributeSensor<Integer> value = Sensors.newIntegerSensor("test.value");
+        AttributeSensor<String> string = Sensors.newStringSensor("test.string");
+        group = app.createAndManageChild(EntitySpec.create(SequenceGroup.class)
+                .configure(SequenceGroup.SEQUENCE_START, 12345)
+                .configure(SequenceGroup.SEQUENCE_INCREMENT, 678)
+                .configure(SequenceGroup.SEQUENCE_VALUE_SENSOR, value)
+                .configure(SequenceGroup.SEQUENCE_STRING_SENSOR, string)
+                .configure(SequenceGroup.SEQUENCE_FORMAT, "0x%04X"));
+        group.setEntityFilter(EntityPredicates.hasInterfaceMatching(".*TestEntity"));
+
+        assertEqualsIgnoringOrder(group.getMembers(), ImmutableSet.of(e1, e2, e3));
+        assertAttributeEquals(e1, value, 12345);
```

Comment:
Can we be sure that `e1`, `e2`, e3` will be the order of the members? Is that because it is called sequentially with the order that the entities were created?